Sources/KituraNet/HTTP/HTTPServer.swift

```diff
@@ -309,15 +318,17 @@ public class HTTPServer: Server {
             }
             .childChannelInitializer { channel in
                 let httpHandler = HTTPRequestHandler(for: self)
+                let serverConfigurationHandler = HTTPServerConfigurationHandler(for: self)
                 let config: NIOHTTPServerUpgradeConfiguration = (upgraders: upgraders, completionHandler: { _ in
                     _ = channel.pipeline.removeHandler(httpHandler)
                 })
                 return channel.pipeline.configureHTTPServerPipeline(withServerUpgrade: config, withErrorHandling: true).flatMap {
                     if let nioSSLServerHandler = self.createNIOSSLServerHandler() {
-                        _ = channel.pipeline.addHandler(nioSSLServerHandler, position: .first)
+                        return channel.pipeline.addHandler(nioSSLServerHandler, position: .first)
                     }
-                    return channel.pipeline.addHandler(httpHandler)
-                }
+                    return channel.eventLoop.makeSucceededFuture(()) } .flatMap {
```

Comment:
you can just drop the `return channel.eventLoop.makeSucceededFuture(()) } .flatMap {`